### PR TITLE
Make interactive snapshot output a bit more obvious

### LINKED ISSUE #1992: Make snapshot more obvious outside of tests
```
Can't compare snapshot to reference when testing interactively.
ℹ Run devtools::test() or `testthat::test_file()` to see changes.
Current value:
Code
  tool_arg(1, letters[1:3], NA)
Condition
  Error:
  ! <elmer::tool_arg> object properties are invalid:
  - @type must be <character>, not <double>
  - @description must be a single string, not a character vector
  - @required must be a TRUE or FALSE, not NA
 ```

->

```
Can't compare snapshot to reference when testing interactively.
ℹ Run devtools::test() or `testthat::test_file()` to see changes.
-- Snapshot result --------------------------------------------------------
Code
  tool_arg(1, letters[1:3], NA)
Condition
  Error:
  ! <elmer::tool_arg> object properties are invalid:
  - @type must be <character>, not <double>
  - @description must be a single string, not a character vector
  - @required must be a TRUE or FALSE, not NA
 -------------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by hadley**
Testing this change revealed a bug in `test_code()`: specifying the default reporter happened _after_ `local_test_context()` meaning that it failed to capture the actual user settings. I fixed this by now requiring a reporter (rather than a backup reporter).

Fixes #1992

@DavisVaughan I can't figure out how to test this, so you might want to run a few snapshot tests interactively (both inside of `test_that()` and directly).

### CODE REVIEW

**Review by DavisVaughan — APPROVED** [review]
I tried running both `test_that()` blocks and `expect_snapshot()` calls directly and all looks good

File: R/snapshot.R [hunk 1]
```diff
@@ -322,12 +322,15 @@ snapshot_accept_hint <- function(variant, file, reset_output = TRUE) {
...
+snapshot_not_available <- function(header, message) {
+  local_reporter_output()
+
   cli::cli_inform(c(
-    "{.strong Can't compare snapshot to reference when testing interactively.}",
-    i = "Run {.run devtools::test()} or {.code testthat::test_file()} to see changes."
+    i = "Can't create snapshot or compare to reference when testing interactively."
   ))
+  cat(cli::rule(header), "\n", sep = "")
   cat(message, "\n", sep = "")
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/snapshot.R]: LGTM!

**Comment by DavisVaughan** [comment]
I think this looks nice now, not sure about your test failure though

Screenshot 2024-11-01 at 10 44 57 AM

**Comment by hadley** [comment]
@DavisVaughan can you take one more look please?

**Review by DavisVaughan — APPROVED** [review]

**Pull request merged**